### PR TITLE
Keep list of active transfers and ensure device widgets show any ongoing transfers

### PR DESCRIPTION
**Pull request by torikulhabib**
Avoid popover no transfer progress

### CODE REVIEW

File: src/Services/ObexManager.vala [hunk 1]
```diff
@@ -20,8 +20,10 @@ public class BluetoothIndicator.Services.ObexManager : Object {
     public signal void transfer_removed (BluetoothIndicator.Services.Obex.Transfer transfer);
     public signal void transfer_active (string address);
     private GLib.DBusObjectManagerClient object_manager;
+    public GLib.HashTable<BluetoothIndicator.Services.Obex.Transfer, string> transferact;
```
Comment (jeremypw) [inline src/Services/ObexManager.vala:23]: I am in favour of using meaningful variable names where possible.  How about `active_transfers`?

Comment (torikulhabib) [reply src/Services/ObexManager.vala:23]: 😃👍

**Comment by jeremypw** [comment]
@torikulhabib Could you show with screenshots how this makes ongoing transfers visible in the wingpanel (which was the subject of the linked issue)? I did not see and change there or any code that would visibly change the indicator.  Or is this a preparatory PR?  Ideally some kind of animation of the indicator would be required I guess.

**Comment by torikulhabib** [comment]
actually this code is more to avoid the absence of progress file transfer takes place. when the computer is first turned on and the bluetooth popover has not been opened. obex_manager is not working because the widget condition is still null.

**Comment by jeremypw** [comment]
Ok, but could you change the description and remove the link to #170 as I do not think it fixes that issue?  If you could give the steps needed to reproduce the issue this PR is fixing that would help.

File: src/Services/ObexManager.vala [hunk 2]
```diff
@@ -20,8 +20,10 @@ public class BluetoothIndicator.Services.ObexManager : Object {
     public signal void transfer_removed (BluetoothIndicator.Services.Obex.Transfer transfer);
     public signal void transfer_active (string address);
     private GLib.DBusObjectManagerClient object_manager;
+    public GLib.HashTable<BluetoothIndicator.Services.Obex.Transfer, string> active_transfers;
 
     construct {
+        active_transfers = new GLib.HashTable <BluetoothIndicator.Services.Obex.Transfer, string> (GLib.str_hash, GLib.str_equal);
```
Comment (tintou) [inline src/Services/ObexManager.vala:26]: active_transfers = new GLib.HashTable <BluetoothIndicator.Services.Obex.Transfer, string> (GLib.direct_hash, GLib.direct_equal);

Comment (torikulhabib) [reply src/Services/ObexManager.vala]: Ok

**Comment by jeremypw** [comment]
@torikulhabib I have pushed some minor code cleanups to this PR - please merge if you are OK with them.

**Comment by jeremypw** [comment]
@torikulhabib Are you able to indicate how to reproduce the problem you are fixing?  As far as I can understand there are conditions under which the popover does not show a file transfer in progress.  However, I have not been able to produce such a state.  When the computer is first turned on there cannot be any file transfers in progress and in master the popover always creates and shows both inbound and outbound transfer progress anyway.

However, I did not find any regressions and it seems sensible to maintain a list of active transfers so I am happy to merge.

**Comment by torikulhabib** [comment]
try transfer file with bluethooth. on progress recieving transfer.
killall io.elementary.wingpanel

**Pull request merged**